### DISCUSSION TITLE
Treat warnings as errors?

### DISCUSSION THREAD

**Discussion by KUGA2** [renovatebot/renovate#32791] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure

### Please tell us more about your question or problem

Hi,

i have seen multiple cases of problems now, that where marked as warnings but my job was ok.

Examples:

```
DEBUG: baseBranches (repository=DevOps/pipeline_templates)
       "baseBranches": ["refs/pull/99926/merge"]
 WARN: Base branch does not exist - skipping (repository=DevOps/pipeline_templates, baseBranch=refs/pull/99926/merge)
```

```
                 "warnings": [
                   {
                     "topic": "**redacted**",
                     "message": "Failed to look up custom.artifactory-generic package **redacted**"
                   }

```

In these examples, a new version that exists was not found. The users would never know a new version existed because the job is just ok.

Is there a configuration setting to change that? Like gcc `-Werror`?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
logLevelRemap

**Reply by KUGA2** [reply]
Thank you.

This setting ([logLevelRemap](https://docs.renovatebot.com/configuration-options/#loglevelremap)) can remap specific messages. I could regex for warning and set it to ERROR or FATAL, But that only works for known messages. Would work for messages like example 2 but not like example 1. Is there a solution for that?

**Reply by rarkins** [reply]
No solution

**Reply by KUGA2** [reply, marked as answer]
I guess i will work around this and parse the log file for "WARN" (but not for "warnings":[]) after renovate ran. And then break the job if it appears.

Thank you for the info!
